src/connectors/rating-menu/connectRatingMenu.js

```diff
@@ -129,31 +132,77 @@ export default function connectRatingMenu(renderFn, unmountFn = noop) {
   checkRendering(renderFn, withUsage());
 
   return (widgetParams = {}) => {
-    const { attribute, max = 5 } = widgetParams;
+    const { attribute, max = 5, step = 1 } = widgetParams;
```

Comment:
I compared https://codesandbox.io/s/instantsearchjs-forked-ttnb0 (previous) and https://codesandbox.io/s/instantsearchjs-forked-92pyp?file=/src/app.js (with this PR) and the number of results changed, even without changing the step. Should we have a default that allows only integer? I'm not sure what step means. If I set it to 1, the warning still shows up, and 2 I think it rounds oddly?

Reply:
The new implementation was keeping the values >= to `max` into the count, while the previous implementation was ignoring them. I fixed the calculation to match the previous implementation.

The `step` parameter changes how the results get visually grouped, for example if you want to display half-stars you can set the step to 0.5.

Grouping with `step` set to `1` (default):

```
>= 4
>= 3
>= 2
>= 1
```

Grouping with `step` set to `0.5`:

```
>= 4.5
>= 4
>= 3.5
>= 3
>= 2.5
>= 2
>= 1.5
>= 1
>= 0.5
```

Changing the grouping does not affect the warning, the only solutions I found to get the correct count (and therefore to fix the warning) is to lower the precision of the values (less possible facet values) and/or to retrieve more facets (to get all possible values).

Reply:
I've updated both sandboxes and the count is now more reasonable, but neither really work as expected (saying ~100 hits, while really there are thousands, in the previous implementation 1 hit).

Since the hits are so different, I think this is a breaking change (unfortunately)

Reply:
I update the implementation to apply the maximum value in the refinement.

eg. you add a ratingMenu with a maximum value of 5 and apply a refinement on 3 and up, you'll get `numericFilters: ["price<=5","price>=3"]`. The count is accurate then.

You can still have some small impressions on the count but I think this is due to engine limitations (to confirm).